### DISCUSSION TITLE
Usage with KeyDB instead of Redis

### DISCUSSION THREAD

**Discussion by aleksasiriski** [authelia/authelia#8215] [Q&A] [answered]
Hello there, I'm using Authelia for a long time now and I really like it, amazing project. I recently started doing more complex things in Kubernetes and encountered an issue with Authelia and KeyDB. For those that don't know, KeyDB is a Redis-compatible KV store that is eventually consistent using async replication (having both multi-masters and active-replicas). For some reason, it seems that after I log into Authelia that my session isn't immediately synced to other KeyDB nodes (my speculation) and that results in some weird bugs.

I workarounded this by utilizing Traefik's sticky sessions to Authelia pods (3 in my case for both Authelia and KeyDB, all masters) and using a service sessionAffinity (ClusterIP) for KeyDB. This gave me a nice static path from client -> authelia pod (always the same one for this session) -> keydb pod (always the same one using authelia's ClusterIP). This results in perfect login attempts when visiting `auth.example.org` directly, but now I'm still left out with another problem: infinite loops caused by Traefik's middleware to check the authelia session cookie.

Now, this ended up being a nice learning experience about Traefik and K8s, but it's still an annoyance. I know that this might be a special usecase and it's not really up to Authelia to fix anything but I was hoping for some guidence in case anyone runs a similar setup. This part is interesting (but not sure wether it's KeyDB only or also supported with Redis): https://docs.keydb.dev/docs/commands/#wait

Thanks for reading.

**Comment by james-d-elliott** [comment, marked as answer]
The WAIT command is used after any write operation to ensure all nodes have the data before continuing. This could be potentially implemented however there are potential deadlocks and other races introduced by adding it. We can look at it during our session rework where we'll take complete control of the session library internally.
